app/controllers/DataSetController.scala

```diff
@@ -178,6 +178,10 @@ class DataSetController @Inject()(userService: UserService,
       uploaderId: Option[String],
       @ApiParam(value = "Optional filtering: List only datasets in the folder with this id")
       folderId: Option[String],
+      @ApiParam(
+        value =
+          "Optional filtering: If a folderId was specified, this parameter controls whether subfolders should be considered, too (default: false)")
+      recursive: Option[Boolean],
```

Comment:
The name `recursive` would be fine if the entire rout would be about searching in folders. But since it is broader, and the arguments can be combined in different combinations, I’d prefer something more explanatory. Hmm, not sure, how about “alsoSearchSubfolders”?

Reply:
Hmm, I think the route param can also be used without searching. For example, one could list all datasets of a folder recursively by providing folderId, recursive=true and no search query. I think that would work (even though it's not used that way). Would you still prefer another name?

Reply:
Yes, but my suggested name might not be the best option. How about “alsoIncludeSubfolders”? or just “includeSubfolders”? My concern is that the bare “recursive” is not going to be associated with the whole folder concept at all

Reply:
Ah, I see. I like `includeSubfolders` :+1: